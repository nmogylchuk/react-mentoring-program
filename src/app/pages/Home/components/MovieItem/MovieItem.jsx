import React from 'react';
import Grid from '@material-ui/core/Grid';
import PropTypes from 'prop-types';
import ThreeDotMenu from "./../../../shared/ThreeDotMenu/ThreeDotMenu";
import './MovieItem.scss';

const MovieItem = ({movie}) => {

    const {poster_path, title, genres, release_date} = movie;
    const releaseDate = new Date(release_date).getFullYear();
    const genre = Array.isArray(genres) ? genres.join(', ') : '';

    return (
        <Grid item lg={3} md={4} sm={6} xs={12}>
            <div className="movie-item">
                <ThreeDotMenu />
                <img className="movie-item__picture" src={poster_path} alt='movie picture'/>
                <div className="movie-item__info">
                    <div className="movie-item__details">
                        <div className="movie-item__title">{title}</div>
                        <div className="movie-item__release-date">{releaseDate}</div>
                    </div>
                    <div className="movie-item__genres">{genre}</div>
                </div>
            </div>
        </Grid>
    );
};

export default MovieItem;

MovieItem.propTypes = {
    poster_path: PropTypes.string,
    title: PropTypes.string,
    genres: PropTypes.array,
    release_date: PropTypes.string
};