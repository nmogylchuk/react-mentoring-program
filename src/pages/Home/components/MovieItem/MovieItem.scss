@import 'src/assets/styles/variables';

.movie-item {
  margin: 10px;
  position: relative;

  &__picture {
    min-height: 470px;
    border: 1px solid $grey;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $grey;
    
    :hover {
      cursor: pointer;
    }
  }

  &__info {
    margin: 10px 0;
  }

  &__details {
    display: flex;
    justify-content: space-between;
    margin-bottom: 10px;
  }

  &__title {
    font-weight: bold;
  }

  &__release-date {
    border: 1px solid grey;
    padding: 3px 10px;
    font-size: 1.2rem;
    height: 2rem;
  }

  &__genres {
    font-size: 1.4rem;
  }
}
