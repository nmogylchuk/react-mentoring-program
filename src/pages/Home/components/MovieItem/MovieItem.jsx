import React, {useState, useCallback} from 'react';
import {useHistory} from 'react-router-dom';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import ThreeDotMenu from 'pages/Home/components/MovieItem/ThreeDotMenu/ThreeDotMenu';
import ThreeDotMenuActions from 'pages/Home/components/MovieItem/ThreeDotMenuActions/ThreeDotMenuActions';
import 'pages/Home/components/MovieItem/MovieItem.scss';

const MovieItem = (props) => {
    const {handleSearchIcon, movie} = props;

    const [threeDotMenuVisible, setThreeDotMenuVisible] = useState(false);

    const handleThreeDotMenuActions = useCallback(() => {
        setThreeDotMenuVisible(!threeDotMenuVisible);
    }, [threeDotMenuVisible]);

    const history = useHistory();

    const goToMovieDetails = () => {
        handleSearchIcon();
        history.push(`/${movie.id}`)
    }

    const releaseDate = new Date(movie.release_date).getFullYear();
    const genre = Array.isArray(movie.genres) ? movie.genres.join(', ') : '';

    return (
        <Grid item lg={3} md={4} sm={6} xs={12}>
            <article className="movie-item">
                <ThreeDotMenu showMenu={handleThreeDotMenuActions}/>
                {threeDotMenuVisible &&
                <ThreeDotMenuActions closeMenu={handleThreeDotMenuActions} movie={movie}/>}
                <img src={movie.poster_path} alt={`picture of ${movie.title}`} onClick={goToMovieDetails}/>
                <div className="movie-item__info">
                    <div className="movie-item__details">
                        <div className="movie-item__title">{movie.title}</div>
                        <div className="movie-item__release-date">{releaseDate}</div>
                    </div>
                    <div className="movie-item__genres">{genre}</div>
                </div>
            </article>
        </Grid>
    );
}

export default MovieItem;

MovieItem.propTypes = {
    movie: PropTypes.shape({
        id: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired,
        poster_path: PropTypes.string.isRequired,
        genres: PropTypes.array.isRequired,
        release_date: PropTypes.string.isRequired
    })
};

