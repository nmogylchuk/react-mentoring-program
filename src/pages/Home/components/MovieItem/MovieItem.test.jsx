import React from 'react';
import rerender from 'react-test-renderer';
import MovieItem from './MovieItem';

describe('when Movie', () => {

    const movie = {
        "id": 447365,
        "title": "Guardians of the Galaxy Vol. 3",
        "tagline": "",
        "vote_average": 0,
        "vote_count": 9,
        "release_date": "2020-05-01",
        "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        "overview": "The third film based on Marvel's Guardians of the Galaxy.",
        "budget": 0,
        "revenue": 0,
        "genres": [
            "Action",
            "Adventure",
            "Science Fiction"
        ],
        "runtime": null
    }

    test('then snapshot created', () => {
        const movieItemComponent = rerender.create(<MovieItem movie={movie}/>);
        const tree = movieItemComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

});