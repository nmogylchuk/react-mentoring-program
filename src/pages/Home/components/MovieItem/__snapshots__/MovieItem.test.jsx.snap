// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Movie then snapshot created 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
>
  <article
    className="movie-item"
  >
    <div
      className="three-dots-menu"
      onClick={[Function]}
    />
    <img
      alt="picture of Guardians of the Galaxy Vol. 3"
      className="movie-item__picture"
      onClick={[Function]}
      src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
    />
    <div
      className="movie-item__info"
    >
      <div
        className="movie-item__details"
      >
        <div
          className="movie-item__title"
        >
          Guardians of the Galaxy Vol. 3
        </div>
        <div
          className="movie-item__release-date"
        >
          2020
        </div>
      </div>
      <div
        className="movie-item__genres"
      >
        Action, Adventure, Science Fiction
      </div>
    </div>
  </article>
</div>
`;
